### PR TITLE
Bugfix/nw23001440/scan with colon

### PR DESCRIPTION
**Pull request by davidepalladino-apuliasoft**
## Description
I tried to fix the problem by checking if the Factor 1 is literal or not. In affirmative case, is built an `StringLiteral` expression to perform the action.
In addition, has been fixed the behavior of `expression.kt` when the number of occurrences is 0.

## Checklist:
- [x] There are tests regarding this feature
- [x] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [x] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt [hunk 1]
```diff
@@ -1618,6 +1618,7 @@ data class ScanStmt(
         } while (index >= 0)
         if (occurrences.isEmpty()) {
             interpreter.setIndicators(this, BooleanValue.FALSE, BooleanValue.FALSE, BooleanValue.FALSE)
+            interpreter.assign(target, IntValue(0))
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt:1621]: Why?

**Review by lanarimarco — CHANGES_REQUESTED** [review]
Can you add into `SCANTEST.rpgle` an example of `SCAN` with column?
Remember that the asserts are in the `executeSCANTEST` function.

**Comment by davidepalladino-apuliasoft** [comment]
> Can you add into `SCANTEST.rpgle` an example of `SCAN` with column? Remember that the asserts are in the `executeSCANTEST` function.

Sure, for next commit. I created a case where the start point is higher than length of word.

**Review by lanarimarco — APPROVED** [review]
Ok

**Pull request merged**